tests/mocha/workspace_svg_test.js

```diff
@@ -116,6 +116,171 @@ suite('WorkspaceSvg', function() {
     });
   });
 
+  suite.only('Viewport change events', function() {
+    function resetEventHistory(eventsFireStub, changeListenerSpy) {
+      eventsFireStub.resetHistory();
+      changeListenerSpy.resetHistory();
+    }
+    function assertSpyFiredViewportEvent(spy, workspace, expectedProperties) {
+      assertEventFired(
+          spy, Blockly.Events.Ui, {element: 'viewport'},
+          workspace.id, null);
+      assertEventFired(spy, Blockly.Events.Ui, expectedProperties,
+          workspace.id, null);
+    }
+    function assertViewportEventFired(eventsFireStub, changeListenerSpy,
+        workspace, expectedEventCount = 1) {
+      var metrics = workspace.getMetrics();
+      var expectedProperties = {
+        element: 'viewport',
+        newValue: {scale: workspace.scale, top: metrics.viewTop,
+          left: metrics.viewLeft}
+      };
+      assertSpyFiredViewportEvent(
+          eventsFireStub, workspace, expectedProperties);
+      assertSpyFiredViewportEvent(
+          changeListenerSpy, workspace,expectedProperties);
+      sinon.assert.callCount(changeListenerSpy, expectedEventCount);
+      sinon.assert.callCount(eventsFireStub, expectedEventCount);
+    }
+    function runViewportEventTest(eventTriggerFunc, eventsFireStub,
+        changeListenerSpy, workspace, clock, expectedEventCount = 1) {
+      clock.runAll();
+      resetEventHistory(eventsFireStub, changeListenerSpy, this.clock);
+      eventTriggerFunc();
+      assertViewportEventFired(
+          eventsFireStub, changeListenerSpy, workspace, expectedEventCount);
+    }
+    setup(function() {
+      defineStackBlock();
+      this.changeListenerSpy = createFireChangeListenerSpy(this.workspace);
+    });
+    teardown(function() {
+      delete Blockly.Blocks['stack_block'];
+    });
+
+    test('inject', function() {
+      // TODO: Address and remove before merging PR
+      // How do we want this to behave?
+      // Change listeners are attached after resizes, but events are disabled
+      // so viewport event is fired, but not listened to (at this time).
+      workspaceTeardown.call(this, this.workspace);
+      var workspace = Blockly.inject('blocklyDiv');
+      this.clock.runAll();
+      assertViewportEventFired(
+          this.eventsFireStub, this.changeListenerSpy, workspace);
+    });
+    test('dispose fires no viewport event', function() {
```

Comment:
Initially seemed like this might be nice add to test, but I'm no longer sure I should add it.